readthedocs/builds/tasks.py

```diff
@@ -206,3 +216,98 @@ def delete_inactive_external_versions(limit=200, days=30 * 3):
                 version.project.slug, version.slug,
             )
             version.delete()
+
+
+@app.task(
+    max_retries=1,
+    default_retry_delay=60,
+    queue='web'
+)
+def sync_versions_task(project_pk, tags_data, branches_data, **kwargs):
+    """
+    Sync the version data in the repo (on the build server).
+
+    Version data in the repo is synced with what we have in the database.
+
+    :param tags_data: List of dictionaries with ``verbose_name`` and ``identifier``.
+    :param branches_data: Same as ``tags_data`` but for branches.
```

Comment:
I think we should refactor these to be a list of named tuples, I think celery should do fine serializing them, but for another PR anyway...

Reply:
This should probably be in a `TODO` in the code, then :)